discovery/src/lib.rs

```diff
@@ -129,7 +138,14 @@ impl Builder {
         )?;
 
         #[cfg(not(feature = "with-dns-sd"))]
-        let svc = libmdns::Responder::spawn(&tokio::runtime::Handle::current())?.register(
+        let _svc = if !_bind_ips.is_empty() {
+            libmdns::Responder::spawn_with_ip_list(&tokio::runtime::Handle::current(), _bind_ips)?
```

Comment:
Is it a Clippy error to just use `self.bind_ips` directly? Seems odd for Clippy to enforce leading underscore style when something is only conditionally unused. 
```suggestion
            libmdns::Responder::spawn_with_ip_list(&tokio::runtime::Handle::current(), self.bind_ips)?
```